### PR TITLE
Finalize notebook gating and harden JAX release checks

### PR DESCRIPTION
**Pull request by conorheins**
## Summary
- wire the notebook manifests into PR and nightly GitHub Actions via `scripts/run_notebook_manifest.py`
- add the remaining release-polish updates for notebook docs, release notes, contributor guidance, and the RC checklist
- harden JAX normalization checks and associative-scan HMM smoothing for CUDA-backed runs

## Testing
- `uv run pytest -n0 -q test/test_utils_jax.py test/test_agent_jax.py test/test_hmm_associative_scan.py`
- `uv run pytest -n0 -q test/test_inductive_inference_jax.py`
- `uv run --group test python scripts/run_notebook_manifest.py test/notebooks/ci_notebooks.txt -- --collect-only`
- `uv run --group test python scripts/run_notebook_manifest.py test/notebooks/nightly_notebooks.txt -- --collect-only`
- `uv run --no-default-groups --extra docs mkdocs build --strict`
- `uv run --with build python -m build --sdist --wheel`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -13,12 +13,12 @@ authors = [
...
-license = {file = "LICENSE"}
+license = "MIT"
+license-files = ["LICENSE"]
 classifiers=[
-    'Development Status :: 4 - Beta',
+    'Development Status :: 5 - Production/Stable',
     'Intended Audience :: Developers',
     'Topic :: Scientific/Engineering :: Artificial Intelligence',
-    'License :: OSI Approved :: MIT License',
     'Programming Language :: Python :: 3',
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pyproject.toml:22]: We removed it for a reason. Modern setuptools is deprecating license classifiers in favor of SPDX-style license = "MIT" plus license-files, and re-adding the classifier just brings back the warning we were explicitly trying to eliminate.

File: setup.cfg [hunk 2]
```diff
@@ -7,12 +7,11 @@ long_description_content_type = text/markdown
 author = Conor Heins, Alexander Tschantz, Tim Verbelen, Dimitrije Markovic
 author_email = conor.heins@gmail.com, tschantz.alec@gmail.com, verbelen.tim@gmail.com, dimitrije.markovic@tu-dresden.de
 license = MIT
-license_file = LICENSE
+license_files = LICENSE
 classifiers =
-    Development Status :: 4 - Beta
+    Development Status :: 5 - Production/Stable
     Intended Audience :: Developers
     Topic :: Scientific/Engineering :: Artificial Intelligence
```
[the remark this answers is not in the document]
Comment (conorheins) [reply setup.cfg:14]: We removed it for a reason. Modern setuptools is deprecating license classifiers in favor of SPDX-style license = "MIT" plus license-files, and re-adding the classifier just brings back the warning we were explicitly trying to eliminate.

File: pymdp/utils.py [hunk 3]
```diff
@@ -104,14 +104,43 @@ def validate_normalization(tensor: Array, axis: int = 1, tensor_name: str = "ten
         Raises `ValueError` if distributions are invalid.
```
[the remark this answers is not in the document]
Comment (conorheins) [reply pymdp/utils.py]: Updated the validate_normalization docstring in utils.py to describe the actual split behavior: eager mode raises ValueError, while traced/JIT mode signals invalid inputs through `eqx.error_if` in [42e9e70](https://github.com/infer-actively/pymdp/pull/373/commits/42e9e7070cae0581c1bec95675beb61dc6edf256)
